applications/SaaS3.md

```diff
@@ -88,64 +86,82 @@ Such reward and punishment mechanism cloud guarantee the miner's loyalty and ser
+   - The three hash values are in agreement. No one will being slashed.
+   - Only two values are in agreement, the check still gains approval. Another miner is found inconsistent and will be slashed on its token stake.
+   - There is no agreement among the three nodes. In this case, the API call returns failure. The gas fee is nonetheless consumed and has no refunds. No party will face slash or claim reward, as neither miner malfunctions nor API bugs can be found attributable. However, the failure-logs are recorded on on-chain storage. The user could raise an issue to the creator to push for a new docker, or lodge a sue claim on SaaS3DAO as a query for potential compensation from miners / creators. For more details, please refer to 9. 3-zkcg.
 
 8. **Court DAO**
-User claim compensation from miners / creators on SaaS3 DAO due to the API failure. The user should provide the fail case details such as the input data and the failed API version (docker image file version). Jury consists of 100 valid users, miners and creators who are active on the DAO events to earn rewards. Jury can download the same version docker image and retest the failure case to determine the failure reason. The on-chain failure log is available for jury members. 
+At Court DAO, the user can pay a fee for sue claims and, subject to the court's ruling, may claim compensation from miners / creators on SaaS3 DAO on grounds of any API failure. To supplement the on-chain failure logs, the user is required to provide the details regarding the failure case, such as the input data and the version of the failed API (docker image file version). The Jury consists of top 100 largest staked creators who are open source developers and active on DAO events. Members of the Jury can download the same version docker image and retest the failure case to investigate the failure cause. The on-chain failure log is available for jury members. 
 
-    Jury will have the following options in terms of on-chain log records and local test results.
+    Jury will have the following options given the on-chain log records and local test results.
 
-    - Reject - nothing is paid to user and sue fee won't be refunded due to user's reason such as fake evidence or unreasonable amount of claims which can be referred to terms and regulation. 
+    - Reject - The user's sue claim is rejected, with no compensation, and the sue fee will not be refunded. Such a ruling may be on the grounds such as forged evidence and unreasonable amount of claims. Terms and conditions and regulations of SaaSDAO apply.
     - Approve
       - Miner malfunction: 
         - Which miners should be slashed. (This is multiple choices refer to on-chain log):
           - Miner1, Miner2, or Miner3
       - Slash the API service in terms of drawbacks or bugs on the service.
-     
+      
+
+9. **3-zkc and gas calculation algorithm (3-zkcg)**
```

Comment:
Your concern is exactly right. To reduce the burden of tokenomics related things, 3-person zk check algorithm (3-zkc) is illustrated at here.